api/resourcediscovery/v1alpha2/resource_discovery.proto

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+import "google/protobuf/any.proto";
+import "google/rpc/status.proto";
+
+package federation.resourcediscovery.v1alpha2;
+
+option go_package = "resourcediscovery/v1alpha2";
+
+// StreamRequest represents a streaming request sent by a federated service mesh
+// consumer to a federated service mesh owner.
+//
+// Following are some of the valid field combinations for a StreamRequest.
+//
+// * Initial request:
+//     resource_url   != ""
+//     response_nonce == ""
+//     status         == nil
+//
+// * ACK, request:
+//     resource_url   != ""
+//     response_nonce != ""
+//     status         != nil
+//       code         == OK
+//       message      == ""
+//
+// * NACK, request:
+//     resource_url   != ""
+//     response_nonce != ""
+//     status         != nil
+//       code         != OK
+//       message      != ""
+message StreamRequest {
+  // REQUIRED. The type URL of the resource being requested.
+  // Example: "type.googleapis.com/federation.types.v1alpha2.FederatedService"
+  string resource_url = 1;
+
+  // The nonce of a consumed StreamResponse to ACK/NACK. This field should be
+  // omitted in the first StreamRequest sent by the federated service mesh
+  // consumer.
+  string response_nonce = 2;
+
+  // The message consumption status of the client. This field should be omitted
+  // in the first StreamRequest sent by the federated service mesh consumer. If
+  // the message is consumed successfully, the status code should be set to OK.
+  // If there is a failure in consuming a message, an appropriate status code
+  // must be set along with the error details in the status' message field.
+  google.rpc.Status status = 3;
+
+  // context which will be used for coorelation between req/response
```

Comment:
It may be good to clarify that correlation is needed because now multiple clients can connect to the same server

Reply:
ok will add some description.